### PR TITLE
maint: Update OpenTelemetry instrumentation packages

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?
Bumps OpenTelemetry instrumentation packages to latest.

## Short description of the changes
- Bump instrumentation pacakages to latest

### CODE REVIEW

**Review by cartermp — APPROVED** [review]
Changes look good, smoky boi not happy though

**Comment by MikeGoldsmith** [comment]
OTel .NET SDK v1.3.2 has been released, we should update to that as part of this PR too.

**Comment by cartermp** [comment]
Blegh, more smoke test failure, but this is ready and fully up to date

**Review by JamieDanielson — COMMENTED** [review]
Running the aspnetcore example app locally and in Docker give an error:

```txt
Unhandled exception. System.TypeLoadException: Could not load type 'Microsoft.Extensions.DependencyInjection.TracerProviderBuilderServiceCollectionExtensions' from assembly 'OpenTelemetry, Version=1.0.0.0, Culture=neutral, PublicKeyToken=7bd6737fe5b67e3c'.

   at Microsoft.Extensions.DependencyInjection.OpenTelemetryServicesExtensions.AddOpenTelemetryTracing(IServiceCollection services, Action`1 configure)

   at Program.<Main>$(String[] args) in /app/examples/aspnetcore/Program.cs:line 11

   at Program.<Main>(String[] args)
```

**Comment by cartermp** [comment]
Good 'ole DI errors with mismatched assembly versions. Should be fixed now.

**Review by JamieDanielson — COMMENTED** [review]
@cartermp beat me to it 🥇 . The one extra package in the example app was old.

Is this something we could or should add in our package? edit for clarity: "this" referring to the `OpenTelemetry.Extensions.Hosting` package.

**Comment by cartermp** [comment]
> Is this something we could or should add in our package?

Can't due to NuGet packaging rules. It's considered pre-release, which would force us back to a pre-release if we took a dependency on it.

**Comment by cartermp** [comment]
Let's wait for 1.4.x to get released. There's some akward states in packages right now that make it not possible for us to upgrade without depending on that chain.

**Comment by cartermp** [comment]
Good news! Updated packages/samples and now we're a bunch of golden little goosies.

Bad news! CI is sad about gpg (pronounced guh-puh-guh) signing in packaging.

**Comment by JamieDanielson** [comment]
> Bad news! CI is sad about gpg (pronounced guh-puh-guh) signing in packaging.

Try a rebase from main, I updated the snupkeg

**Comment by cartermp** [comment]
fyi I am still working on the commoninsteumenttations package updates. Some msbuild fuckery is needed because of .NET 7

File: src/Honeycomb.OpenTelemetry.CommonInstrumentations/Honeycomb.OpenTelemetry.CommonInstrumentations.csproj [hunk 1]
```diff
@@ -26,31 +26,30 @@
     <RepositoryType>git</RepositoryType>
   </PropertyGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And
+                        $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '6.0'))">
```
Comment (cartermp) [inline src/Honeycomb.OpenTelemetry.CommonInstrumentations/Honeycomb.OpenTelemetry.CommonInstrumentations.csproj:30]: I know, this is software gore, but it's the right way to do it.

Example: https://github.com/dotnet/sdk/blob/ccd25c39368a4f9ed8c80a2b47251f4fb1326417/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets#L28-L33

**Pull request merged**